src/Markup/Avalonia.Markup.Xaml/XamlIl/CompilerExtensions/Transformers/AvaloniaXamlIlControlTemplateTargetTypeMetadataTransformer.cs

```diff
@@ -25,7 +25,7 @@ public IXamlIlAstNode Transform(XamlIlAstTransformationContext context, IXamlIlA
             
             if ((tt?.Values.FirstOrDefault() is XamlIlTypeExtensionNode tn))
             {
-                targetType = tn.Type;
+                targetType = tn.Value;
```

Comment:
This indeed seems like a typo/oversight, since `{x:Type}`'s `Type` is always `System.Type`